### ISSUE TITLE
Godot headless builds 3.0.1 / 3.0.2 Docker

### ISSUE THREAD

**Issue by herman-rogers** [godotengine/godot#18257]
Godot version 3.0.1, 3.0.2

Headless server binary (running in Ubuntu Docker 16.04)

Getting multiple issues when exporting my game via headless mode in version 3.0.1 and 3.0.2. I have it working with Godot version 2.1.4.

Essentially I'm running this command:
```godot --export --quit 'Mac OSX' /build/game.dmg --path /project/directory -v```

My first issue, is that I can't find any documentation to place the export templates. Since this is headless and in Docker I must place them manually - I cannot open the GUI and install that way. I have placed them in ~/.godot/templates/ (which seems to be deprecated) and in ~/.local/share/godot/templates. I am unsure if I need to place them in a directory called 3.0.x.mono.stable or similar. I also get different errors or issues with each platform:

'Mac OSX': no output file, the build info I get is: 

```
load resource: /root/.confg/godot/editor_settings-3.tres
EditorSettings: Load OK!
FPS: 1
EditorSettings: Save OK!

root@12315213:/#
```

'Linux X11' (I've also tried Linux/X11): ERROR: fs_changed: Unknown export preset

'Windows Desktop':
```
Condition 'err' is true. returned: err
  At: core/os/dir_access.cpp: 305.
EditorSettings Save OK!
```

Essentially I'm not sure where to put the export templates (cannot find documentation on this - only a few forum posts) and if my command I'm running is 100% correct or if I'm missing some vital things. I have also read that headless server builds is not supported, but in the release notes of 3.0.1 it officially says it works like it does in 2.1 so I believe this should work fine, its just the config I'm messing up.

As a side note, if anyone can provide information that'd be helpful - I'm also looking to open source our docker images for Godot and I'll be happy to document how to do headless server builds correctly within the Readme of that project.

**Steps to reproduce:**

Unzip the attached file, cd into Universal/ run ```docker-compose up``` and then ssh into the container by running ```docker exec -it godot-universal-3.0.1 bash``` and then run any export command you fancy.

a) Templates live in ~/.local/share/godot/templates/templates
b) There is a test game in /context/BloodFortress
c) godot is added to the PATH so you can run it from anywhere.

[Universal.zip](https://github.com/godotengine/godot/files/1920769/Universal.zip)

**Minimal reproduction project:**
Just run any export.

**Comment by herman-rogers** [comment]
#16949 This is a related issue but its not 100% exactly the same.

**Comment by herman-rogers** [comment]
So I just figured out that --export-debug doesn't work, but --export_debug does work. I've pasted the output below and it seems as if maybe the export templates are just in the wrong place. Does anyone have info where the right folder is? I've tried ~/.local/share/godot/templates/3.0.1.stable.mono/ but doesn't seem to work.

<img width="512" alt="image" src="https://user-images.githubusercontent.com/3153479/38885043-8e6190fa-423f-11e8-8eed-da839feb54d4.png">

**Comment by herman-rogers** [comment]
Ok so the above comment was a red herring - ```--export-debug``` does work as expected, using ```--export_debug``` causes a strange error where templates supposedly aren't found. But really its just the command failing. Running ```godot --export-debug --build-solutions --quit "Mac OSX" /target/location/ --path /path/to/project/``` seems to be working. At the moment the .dmg is corrupted so its not 100% at the moment but at least its compiling. I also discovered that inside the export templates theres a version.txt with a version name. Templates must be placed inside the folder with this version name to be found correctly. I.e. ~/.local/share/godot/templates/3.0.1.stable/. Still need to investigate as to why the Mac build is corrupted.

**Comment by herman-rogers** [comment]
So basically the command needs to be structured as:

 ```godot --export-debug "Mac OSX" /target/location/ --path /path/to/project/ --build-solutions --quit -v```

For OSX the target location also needs to specify a file and that file needs to be a .zip, otherwise the build will be corrupted.

This can probably be closed - its more of an issue with lack of documentation rather than something not fundamentally working.

**Comment by vayan** [comment]
Hey @herman-rogers! I'm using your docker image to export a project but I'm still getting some `Unknown export preset: Mac OSX` (same error for other presets)

I have them under `~/.local/share/godot/templates/3.0.2.stable/` but Godot doesn't seem to find them 🤔 
Do you have any non-committed tricks laying around? just in case 😄

**Comment by akien-mga** [comment]
@vayan You have the templates installed, but the error is about missing *presets*. Those are configured for each project individually in the editor, to define what platforms to export to and how. The name "Mac OSX" can be defined manually.

**Comment by vayan** [comment]
Right! I'm stupid, was I too focused on finding the right path for the templates 😄 thanks @akien-mga my CI is working now.
